engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlanConfiguration.java

```diff
@@ -124,7 +157,744 @@ public static StramElement fromValue(String value) {
+      }
+
+      if (!ContextUtils.CONTEXT_CLASSES.equals(confElementContextClasses)) {
+        throw new IllegalStateException("All the context classes "
+                                        + ContextUtils.CONTEXT_CLASSES
+                                        + " found in "
+                                        + Context.class
+                                        + " are not used by ConfElements "
+                                        + confElementContextClasses);
+      }
+    }
+
+    /**
+     * This is a recursive method to initialize the ambiguous elements for each
+     * {@link ConfElement}.
+     *
+     * @param element The current {@link ConfElement} at which to start initializing
+     * the ambiguous elements.
+     * @return The set of all simple attribute names encountered up to this point.
+     */
+    public static Set<String> setAmbiguousAttributes(ConfElement element)
+    {
+      Set<String> ambiguousAttributes = Sets.newHashSet();
+      Set<String> allChildAttributes = Sets.newHashSet(element.getContextAttributes());
+
+      for (ConfElement childElement: element.getChildren()) {
+        Set<String> allAttributes = setAmbiguousAttributes(childElement);
+        ambiguousAttributes.addAll(childElement.getAmbiguousAttributes());
+
+        @SuppressWarnings("unchecked")
+        Set<String> intersection = (Set<String>)Sets.newHashSet(CollectionUtils.intersection(allChildAttributes,
+                                                                                             allAttributes));
+        ambiguousAttributes.addAll(intersection);
+        allChildAttributes.addAll(allAttributes);
+      }
+
+      element.setAmbiguousAttributes(ambiguousAttributes);
+      element.setAllChildAttributes(allChildAttributes);
+
+      return allChildAttributes;
+    }
+
+    private final StramElement element;
+    private final ConfElement parent;
+    private Set<ConfElement> children = Sets.newHashSet();
+    private final Set<StramElement> allRelatedElements = Sets.newHashSet();
+    private final Class<? extends Context> contextClass;
+    private Set<String> ambiguousAttributes = Sets.newHashSet();
+    private Set<String> contextAttributes = Sets.newHashSet();
+    private Set<String> allChildAttributes = Sets.newHashSet();
+
+    /**
+     * This creates a {@link ConfElement}.
+     *
+     * @param element The current {@link StramElement} representing a {@link ConfElement}.
+     * @param parent The parent {@link ConfElement}.
+     * @param additionalRelatedElements Any additional {@link StramElement} that could be
+     * related to this {@link ConfElement}.
+     * @param contextClass The {@link Context} class that contains all the attributes to
+     * be used by this {@link ConfElement}.
+     */
+    ConfElement(StramElement element,
+                ConfElement parent,
+                Set<StramElement> additionalRelatedElements,
+                Class<? extends Context> contextClass)
```

Comment:
Currently they cannot. Ports can only belong to operators, operators can only belong to apps.